sycl/doc/extensions/USM/USM.adoc

```diff
@@ -299,26 +331,36 @@ Parameters::
   * `size_t count` - number of bytes to copy
 Return value:: Returns an event representing the copy operation.
 
-===== memset
+===== memfill
 [source,cpp]
 ----
 class handler {
  ...
  public:
   ...
-  void memset(void* ptr, int value, size_t count);
+  (1)
+  void memfill(void* ptr, const void* pattern, size_t pattern_size, size_t count);
+  (2)
+  template <typename T>
+  void memfill(void* ptr, const T* pattern, size_t count)
 };
 
 class queue {
  ...
  public:
   ...
-  event memset(void* ptr, int value, size_t count);
+  (1)
+  event memfill(void* ptr, const void* pattern, size_t pattern_size, size_t count);
+  (2)
+  template <typename T>
+  void memfill(void* ptr, const T* pattern, size_t count);
```

Comment:
Will this overload work if `T` is not trivially copyable?  If we added a check here it might help users avoid making mistakes.  Same question/suggestion for `memset` if you add it back in.

Reply:
We should probably specify.  The buffer version states: `Replicates the value of src into the memory object accessed by dest. T must be a scalar value or a SYCL vector type.`

Trivially copyable is probably a better restriction.

Reply:
Can we make `pattern` a `const T&`?

Does `count` have to be a multiple of `sizeof(T)`? I guess that it should, otherwise we have an incomplete object or uninitialized memory at the end.

In that sense, a signature like that would probably be safer:

```c++
template <typename ForwardIterator, typename T>
void memfill(ForwardIterator dest_first, ForwardIterator dest_last, const T& pattern);
```

where `std::iterator_traits<ForwardIterator>::value_type` is `char`, `unsigned char` or convertible to `T`.

This is consistent with `std::uninitialized_fill` and makes it  An implementation can still hardware accelerate this as long as `ForwardIterator` also satisfies `LegacyContiguousIterator` and the value type is `T`.

Reply:
Changed to a reference.